api/src/main/java/jakarta/mail/internet/MimePartDataSource.java

```diff
@@ -149,9 +159,7 @@ public String getName() {
      * @since JavaMail 1.1
      */
     @Override
-    public synchronized MessageContext getMessageContext() {
-	if (context == null)
-	    context = new MessageContext(part);
-	return context;
+    public MessageContext getMessageContext() {
```

Comment:
why was `synchronized` removed?

Reply:
There is no point to have it synchronized because the context is instanced now in the constructor. It is not possible that 2 threads obtain a different instance by calling that method.